salesforcecli/plugin-source
# Sm/bundle-delete-nuts

_draft as it requires https://github.com/forcedotcom/source-tracking/pull/55_

### What does this PR do?
fixes json output formatting for push commands (didn't match toolbelt)
new option in sfdx-project.json for pushPackageDirectoriesSequentially
use new STL logic for byPkgDir
iterate array of componentSets for push
NUT for asserting how many deploys happened from the org
NUTs for LWC bundle-member-deletion @W-10216279@

What issues does this PR fix or reference?
@W-10146239@ 

---

QA notes:

top-level boolean prop pushPackageDirectoriesSequentially on sfdx-project.json

use this updated push command to QA the STL changes
I used https://github.com/salesforcecli/sample-project-multiple-packages as a test repo

check what happens with duplicate files in json output from subsequent deployments--do you get duplicates? Is that different from toolbelt or the same?

testing ideas that I didn't try
examining hook output (should fire once for each directory)
check source tracking files (source:status) on an mpd deployment
check pushPackageDirectoriesSequentially when there's only one pkgDirectory
test the human (non-json) output with a MPD project...the messages and fileResponses should all merge into one table across all the deploys
have a failure on any one pkgDir...it should stop any other deployments from happening.
